src/modules/dashboard/components/ColonyHome/ColonyExtensions/ColonyExtensions.tsx

```diff
@@ -29,41 +33,51 @@ const ColonyExtensions = ({ colony: { colonyName, colonyAddress } }: Props) => {
   });
 
   if (loading) {
-    return <SpinnerLoader />;
+    return (
+      <div className={styles.main}>
+        <Heading appearance={{ size: 'normal', weight: 'bold' }}>
+          <FormattedMessage {...MSG.title} />
+        </Heading>
+        <SpinnerLoader />
+        <span className={styles.loadingText}>
+          <FormattedMessage {...MSG.loadingData} />
+        </span>
+      </div>
+    );
   }
 
   return (
     <div className={styles.main}>
-      <Heading appearance={{ size: 'normal', weight: 'bold' }}>
-        <FormattedMessage {...MSG.title} />
-      </Heading>
-      {data?.processedColony && !loading ? (
-        <ul>
-          {data.processedColony.installedExtensions
-            .filter(
-              (extension) =>
-                extension.details.initialized &&
-                !extension.details.missingPermissions.length,
-            )
-            .map((extension) => {
-              const { address, extensionId } = extension;
-              return (
-                <li key={address} className={styles.extension}>
-                  <NavLink
-                    className={styles.invisibleLink}
-                    to={`/colony/${colonyName}/extensions/${extensionId}`}
-                    text={extensionData[extensionId].name}
-                  />
-                  <ExtensionStatus
-                    installedExtension={extension}
-                    deprecatedOnly
-                  />
-                </li>
-              );
-            })}
-        </ul>
-      ) : (
-        <SpinnerLoader />
+      {data?.processedColony && (
```

Comment:
Please include the `main` div as well in your logic check.

So something like;

`return data?.processedColony ? <div className="main">...</div> : null`